_episodes/09-challenges.md

```diff
@@ -13,31 +13,153 @@ keypoints:
 In this episode, we will provide two different challenges for you to attempt,
 based on the skills you have acquired so far. One of the challenges will be
 related to the shape of objects in images (*morphometrics*), while the other 
-will be related to colors of objects in images (*colorimetrics*). We will not
-provide solution code for either of the challenges, but your instructors should
-be able to give you some gentle hints if you need them.
+will be related to colors of objects in images (*colorimetrics*).
 
 ## Morphometrics: Bacteria Colony Counting
 
 As mentioned in the workshop [introduction]({{ page.root }}/01-introduction), 
 your morphometric challenge is to determine how many bacteria colonies are in 
-each of these images. These images can be found in the 
-**Desktop/workshops/image-processing/10-challenges/morphometric** directory. 
+each of these images: 
 
 ![Colony image 1](../fig/00-colonies01.jpg)
 
 ![Colony image 2](../fig/00-colonies02.jpg)
 
 ![Colony image 3](../fig/00-colonies03.jpg)
 
-Write a Python program that uses skimage to count the number of bacteria
-colonies in each image, and for each, produce a new image that highlights the colonies.
-The image should look similar than this one:
-
-![Sample morphometric output](../fig/00-colony-mask.png)
-
-Additionally, print out the number of colonies for each image.
-
+> ## Morphometrics for bacterial colonies
+> 
+> Write a Python program that uses skimage to count the number of bacteria
+> colonies in each image, and for each, produce a new image that highlights the colonies.
+> The image should look similar to this one:
+> 
+> ![Sample morphometric output](../fig/09-colonies01-summary.png)
+> 
+> Additionally, print out the number of colonies for each image.
+> 
+> Use what you learnt in the [histograms]({{ page.root }}/05-creating-histograms), 
+> [thresholding]({{ page.root }}/07-thresholding) and 
+> [Connected Component]({{ page.root }}/08-connected-components) episodes.
+> 
+> > ## Solution
+> > 
+> > First, let's work through the process for one image: 
+> > ~~~
+> > import numpy as np
+> > import skimage.filters
+> > import skimage.color
+> > import skimage.io
+> > import matplotlib.pyplot as plt
+> > %matplotlib widget
+> >
+> > bacteria_image = skimage.io.imread("00-colonies01.jpg")
+> > 
+> > # display the image
+> > fig, ax = plt.subplots()
+> > plt.imshow(bacteria_image)
+> > plt.show()
+> > ~~~
+> > {: .language-python}
+> > ![Colony image 1](../fig/00-colonies01.jpg)
+> > Next, we need to threshold the image to create a mask that covers only
+> > the dark bacterial colonies. This is easier using a grayscale
+> > image, so we convert it here:
+> > ~~~
+> > gray_bacteria = skimage.color.rgb2gray(bacteria_image)
+> > 
+> > # display the gray image
+> > fig, ax = plt.subplots()
+> > plt.imshow(gray_bacteria, cmap='gray')
+> > plt.show()
```

Comment:
Did you want to include the grayscale image on the webpage here?

Reply:
Good point @uschille - I'll add this.